src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double RESIZE_FACTOR = 1.5;
+
+    private Object[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            resize();
+        }
```

Comment:
You call resize method only in combination with if block. It's better to put this if inside method
```suggestion
        resizeIfRerquired();
```